### PR TITLE
add fullscreen option to Modal component

### LINKED ISSUE #14: Update Modal -
- Vertically centered  https://v5.getbootstrap.com/docs/5.0/components/modal/#vertically-centered
- fullscreen https://v5.getbootstrap.com/docs/5.0/components/modal/#fullscreen-modal

### PR DESCRIPTION
**Pull request by phwebi**
This PR adds a `fullscreen` option to the `Modal` component. The vertically centered option is already implemented with the `centered` prop.

Closes https://github.com/gthomas-appfolio/reactstrap/issues/14

### CODE REVIEW

File: types/lib/Modal.d.ts [hunk 1]
```diff
@@ -25,6 +25,7 @@ export interface ModalProps extends React.HTMLAttributes<HTMLElement> {
   backdropTransition?: FadeProps;
   modalTransition?: FadeProps;
   centered?: boolean;
+  fullscreen?: boolean | 'sm' | 'md' | 'lg' | 'xl';
```
Comment (gthomas-appfolio) [inline types/lib/Modal.d.ts:28]: Are we missing an `xxl` option?  Or is that just `true`?

File: docs/lib/Components/ModalsPage.js [hunk 2]
```diff
@@ -113,6 +121,33 @@ const ModalsPage = () => {
         </PrismCode>
       </pre>
 
+      <h4>Fullscreen Modals</h4>
+
+      <div className="docs-example">
+        <div className="btn-group">
+          <div className="btn">
```
Comment (gthomas-appfolio) [inline docs/lib/Components/ModalsPage.js:128]: I don't think `btn` alone is valid, and/or should we use Button and ButtonGroup?  Is this style copied from other docs?

File: types/lib/Modal.d.ts (hunk 1 above)
Comment (zldavis) [reply types/lib/Modal.d.ts:28]: we haven't added `xxl` to anything yet. This is probably the only conflict with those changes, so might as well get this PR in first and can go back and add `xxl` to it in that change.

Comment (phwebi) [reply types/lib/Modal.d.ts:28]: I also didn't see the xxl option in the modal documentation 🤔

File: docs/lib/Components/ModalsPage.js (hunk 2 above)
Comment (phwebi) [reply docs/lib/Components/ModalsPage.js:128]: yeah - I copied from another section on the page 😂 same format as the rest of the sections

File: types/lib/Modal.d.ts (hunk 1 above)
Comment (zldavis) [reply types/lib/Modal.d.ts:28]: oh yeah, there doesn't appear to be an `xxl` [modal size](https://v5.getbootstrap.com/docs/5.0/components/modal/#fullscreen-modal)

**Review by gthomas-appfolio — APPROVED** [review]

**Pull request merged**